dev/io.openliberty.data.internal.persistence/src/io/openliberty/data/internal/persistence/RepositoryImpl.java

```diff
@@ -421,24 +422,30 @@ private NonUniqueResultException excNonUniqueResult(QueryInfo queryInfo,
      * @return exception to replace with, if any. Otherwise, the original.
      */
     @Trivial
-    static RuntimeException failure(Exception original) {
+    static RuntimeException failure(Exception original, EntityManagerBuilder emb) {
         final boolean trace = TraceComponent.isAnyTracingEnabled();
         RuntimeException x = null;
         if (original instanceof PersistenceException) {
             for (Throwable cause = original; x == null && cause != null; cause = cause.getCause()) {
                 if (trace && tc.isDebugEnabled())
                     Tr.debug(tc, "checking " + cause.getClass().getName() + " with message " + cause.getMessage());
 
-                // TODO If this ever becomes real code, it should be delegated to the JDBC integration layer
-                // where there is more thorough logic that takes into account configuration and database differences
-                if (cause instanceof SQLRecoverableException
-                    || cause instanceof SQLNonTransientConnectionException
-                    || cause instanceof SQLTransientConnectionException)
-                    x = new DataConnectionException(original);
-                else if (cause instanceof SQLSyntaxErrorException)
-                    x = new MappingException(original);
-                else if (cause instanceof SQLIntegrityConstraintViolationException)
-                    x = new EntityExistsException(original);
+                if (emb != null && cause instanceof SQLException) { //attempt to have the JDBC layer determine if this is a connection exception
```

Comment:
We know it will never work for other types, so we might as well avoid those errors by checking the instance,
```suggestion
                if (emb instanceof DBStoreEMBuilder && cause instanceof SQLException) { //attempt to have the JDBC layer determine if this is a connection exception
```